crates/matrix-sdk/src/event_cache/room/events.rs

```diff
@@ -158,6 +175,18 @@ impl RoomEvents {
         self.chunks.items()
     }
 
+    /// Get all updates from the room events as [`VectorDiff`].
+    ///
+    /// Be careful that each `VectorDiff` is returned only once!
+    ///
+    /// See [`AsVector`] to learn more.
+    ///
+    /// [`Update`]: matrix_sdk_base::linked_chunk::Update
+    #[allow(unused)] // gonna be useful very soon! but we need it now for test purposes
+    pub fn updates_as_vector_diffs(&mut self) -> Vec<VectorDiff<Event>> {
+        self.chunks_updates_as_vectordiffs.take()
```

Comment:
Why do we have to store a `chunks_updates_as_vectordiffs` field, and why is not sufficient to expose the underlying method? Now that the underlying linked chunk is properly reset and not re-created, the stream won't close after a reset, right?

Reply:
This is not a `Stream` :-). If we recreate the `AsVector` everytime, it will lose all its states, and the `VectorDiff` would be pretty messy.

Reply:
Would we have to re-create an `AsVector`? I'm not sure about that.

Imagine we had:

```rust
pub fn as_vector(&self) -> ... {
  self.chunks.as_vector()
}
```

The resulting `AsVector` could live as long as the caller wants, and it would always be sync'd with the underlying linked chunk, as long as it lived. In that case, we wouldn't have to store anything new in the linked chunk. Is there any reason we cannot do this?